src/lib/internal_tracing/internal_tracing.mli

```diff
@@ -0,0 +1,158 @@
+     - ["@produced_block_state_hash"]: must be issued when the state hash of a
+       produced block is known.
+
+    Internally, these control commands are issued too:
+
+     - ["@current_block"]: used to notify of a execution context change that
+       brings a different block into context.
+     - ["@current_call_id"]: used to notify of a execution context change that
+       brings a different concurrent verifier or prover call into context.
+     - ["@internal_tracing_enabled"]: issued whenever internal tracing is enabled.
+     - ["@internal_tracing_disabled"]: issued whenever internal tracing is disabled.
+     - ["@mina_node_metadata"]: associates global metadata about the current node
+       (useful for version, branch, etc)
+     - ["@rotated_log_end"]: issued when a log file is rotated to notify
+       log consumers.
+     - ["@rotated_log_started"]: issued when a log file is rotated to notify
+       log consumers.
+
+    {1 Output Format}
+
+    The output format is {{: https://jsonlines.org/} JSON Lines}.
+
+    Checkpoints are represented as arrays of two elements, the tag and a timestamp:
+
+    Control commands are represented as JSON objects, with one key and one value.
+    The key is the control command name, and the value is the parameter.contents
+
+    {2 Example}
+
+    {[
+     {"current_block":"3NKk9rxqKp2fyt9fX4JyTu6yRMDfk3k7Wke2aDZVvQs7ReDazxbZ"}
+     ["External_block_received",1677161355.688698]
+     {"block_metadata":{"blockchain_length":"2"}}
+    ]}
+
+    *)
+
+(** {1 API} *)
+
+(** [is_enabled ()] returns [true] if internal tracing is enabled, and [false] otherwise. *)
+val is_enabled : unit -> bool
+
+(** [register_toggle_callback callback] will register [callback] to be called whenever
+    internal tracing is toggled.
+
+    This is useful to syncronize internal tracing done by subprocesses like the verifier
+    and prover.
+
+    [callback] will be called with [true] if internal tracing must be enabled, and with
+    [false] if it must be disabled. It must return a [Deferred.t] that will be resolved
+    once the call completes. *)
+val register_toggle_callback : (bool -> unit Async_kernel.Deferred.t) -> unit
+
+(** [toggle `Enabled] will enable tracing.
+    [toggle `Disabled] will disable tracing.
+
+    If [force] is [false] (the default), and if tracing is already active,
+    then trying to enable tracing is a noop.
+
+    The returned promise will be resolved when all the calls to the registered toggle
+    callbacks have been resolved. *)
+val toggle :
+     logger:Logger.t
+  -> ?force:bool
+  -> [ `Enabled | `Disabled ]
+  -> unit Async_kernel.Deferred.t
+
+(** [with_state_hash state_hash f] runs [f] in a context in which checkpoints
+    and metadata will be associated to a block with state hash equal to [state_hash].
+
+    Any context in which checkpoints or metadata are recorded must be wrapped
+    in either [with_state_hash] or [with_slot] for the checkpoints to be
```

Comment:
```suggestion
    in either {!val:with_state_hash} or {!val:with_slot} for the checkpoints to be
```
if you want to enable links (see Sec. 2.4.2 in https://v2.ocaml.org/manual/ocamldoc.html#s%3Aocamldoc-comments). Might be superfluous since the 2 functions here are next to each other.